### PR TITLE
Adds campaign banner

### PR DESCRIPTION
**Pull request by rosschapman**
This PR adds a banner to top of page for ad-like things, such as promoting a fundraising campaign. The user can close the banner which will set a cookie to hide the banner for 1 day (because why not remind them the next day of awesome things!).

Looks like:

menubar_and_techtonica

techtonica

### CODE REVIEW

File: static/css/style.css [hunk 1]
```diff
@@ -133,7 +137,7 @@ button {
 }
 
 .main-content {
-  margin-top: 4rem;
+  margin-top: 5rem;
```
Comment (rosschapman) [inline static/css/style.css]: The main content area was actually getting cut off.

**Comment by rosschapman** [comment]
@MichelleGlauser black background?  Also what if we just keep this PR open until you come up with copy for the banner.

File: templates/base.html [hunk 2]
```diff
@@ -71,5 +71,28 @@
     <script type="text/javascript" src="{{ url_for('static', filename='js/index.js') }}"></script>
     <script type="text/javascript" src="https://s3.amazonaws.com/downloads.mailchimp.com/js/signup-forms/popup/embed.js" data-dojo-config="usePlainJson: true, isDebug: false"></script>
     <script type="text/javascript"></script>
```
Comment (rosschapman) [inline templates/base.html]: Looks like extra <script> tag here

File: templates/base.html [hunk 3]
```diff
@@ -59,5 +71,28 @@
...
+    <script type="text/javascript" language="javascript">
+       $(document).ready(function() {
+            function numberWithCommas(x) {
+              return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+            }
+            $.getJSON("http://api.indiegogo.com/1/campaigns/14952534#.json?api_token=060335a95d69355cf5e3bef25c36d1be61c8389c3d32c54b402fd5034304df9d", function(gogodata) {
```
Comment (rosschapman) [inline templates/base.html]: @ErinOC possible security issue with the token exposed client-side? What about rendering on server.

**Comment by rosschapman** [comment]
@ErinOC + @MichelleGlauser cool to see more work on this! Let me know if I can help. I did a quick unsolicited code review and left a couple comments. Hope that's ok. 😳

**Comment by MichelleGlauser** [comment]
Thanks, @rosschapman, those are good points about security and repetition. I know that @ErinOC is unable to do any more work on this, unfortunately. Would you mind seeing what you can do, Ross? Or maybe @moagm316, since you've been updating the site most recently, maybe you can take a look or implement what Ross does this evening? The sooner we get this banner up, the better for the campaign.

**Comment by moagm316** [comment]
Sure if you approve the commit I can do the git push this evening.

**Comment by rosschapman** [comment]
@MichelleGlauser double checked, security is not an issue since Indiegogo's API only allows GET (safe) requests and Erin implemented this totally according to the documentation. 

I've resolved the merge conflicts, but this is not ready to go. Current visual state is not great:
techtonica

Not sure if the styling just wasn't completed yet? Even if I check out to Erin's commit 674c594 it still looks like this. 

There also seems to be some problem with the API token or campaign ID because I'm getting a 404 when trying to hit the API. Can you confirm the API token and campaign ID please?

**Comment by MichelleGlauser** [comment]
@rosschapman, thanks for looking at this early. @ErinOC said something in [her pull request](https://github.com/MichelleGlauser/techtonica/pull/18) about the original banner not being in Sass, so she only committed the Sass files because the CSS overrode the Sass and showed weird results—is that original CSS still in there and overriding?

**Comment by rosschapman** [comment]
Oh that might be it. I don't think I was aware we were compiling sass at the time and added styling directly to style.css (whoops). I think I can resolve this actually. I may have some time later to take a stab at it.

**Comment by rosschapman** [comment]
@MichelleGlauser ok I have a basic banner whipped into shape for ya. Here are some screenshots:

**Smaller device**

techtonica

**Desktop**
menubar_and_techtonica

Also note that I've removed the ability to hide the banner. I mean, it's important! 
Also note that generally the CSS is not in the most maintainable state, but would be a nice project for someone to clean up and define a clearer methodology and architecture.

**Comment by MichelleGlauser** [comment]
Hmm, it looks kind of weird on mine—not centered, etc. I'm guessing that's just because the progress bar is missing. Could you please re-enable the Indiegogo API stuff and use JSONP so you don't get the cross-domain error? 'JSONP—If the URL includes the string "callback=?" (or similar, as defined by the server-side API), the request is treated as JSONP instead. See the discussion of the jsonp data type in $.ajax() for more details.' [http://api.jquery.com/jQuery.getJSON/](http://api.jquery.com/jQuery.getJSON)/

**Pull request merged**